embedchain/embedchain.py

```diff
@@ -455,12 +455,14 @@ def retrieve_from_database(self, input_query: str, config: Optional[BaseLlmConfi
 
             db_query = ClipProcessor.get_text_features(query=input_query)
 
-        contents = self.db.query(
+        data = self.db.query(
             input_query=db_query,
             n_results=query_config.number_documents,
             where=where,
             skip_embedding=(hasattr(config, "query_type") and config.query_type == "Images"),
         )
+
+        contents = list(map(lambda x: x[0], data)) if len(data) > 0 and isinstance(data[0], tuple) else data
```

Comment:
This is quite complicated and not easy to understand. You may want to simplify this.

Reply:
okay, simplified the code.